CMakeLists.txt

```diff
@@ -1,8 +1,16 @@
 cmake_minimum_required(VERSION 3.5)
```

Comment:
Weirded out because how can we have these issues if minimum is 3.5 in the first place :-) ?

Reply:
Hi Frank! I'm here because I also saw gtsam [failing](https://build.ros2.org/job/Hdev__gtsam__ubuntu_jammy_amd64/141/console) on ROS build farms :-) 

The point is that, for example CMP0082 is old, but 156, etc. require [cmake >=3.29](https://cmake.org/cmake/help/latest/policy/CMP0156.html) (!) :-) 

It's cool having them, but with these protections keep working on old versions.

Reply:
We are still supporting Ubuntu 20.04, which has cmake 3.16. Since 3.16 > 3.5 this will satisfy cmake_minimum_required(VERSION 3.5), but (see issue #1819) cmake 3.16 does not yet know about the new policies since they were introduced with later versions, and hence it fails.
You would have to bump the version requirement to cmake 3.30 (!!)

Reply:
Dang it, my internet is super flakey so this (from this morning) did not go through. You beat me to it @jlblancoc .

Reply:
Ah, you know, I was being dense: I read 3.5 as > 3.16, interpreting them as floats :-) thanks for merging!